src/main/java/me/arminb/sara/dao/QuestionDAOImpl.java

```diff
@@ -0,0 +1,181 @@
+            if (cursor.hasNext()) {
+                Document questionDoc = cursor.next();
+                Question question = mapQuestion(questionDoc);
+                return question;
+            }
+            else{
+                return null;
+            }
+        } catch (MongoException e) {
+            logger.warn("Failed to fetch question {} from the database", id, e);
+            throw new DataAccessException();
+        }
+    }
+
+    @Override
+    public List<Question> find(String title, String tag, Integer pageNumber, Integer pageCount) throws DataAccessException {
+        if (pageNumber == null){
+            pageNumber = DEFAULT_PAGE_NUMBER;
+        }
+        if (pageCount == null){
+            pageCount = DEFAULT_PAGE_COUNT;
+        }
+        List<Question> list = new ArrayList();
+        try {
+            MongoCollection<Document> collection = database.getCollection("questions");
+            BasicDBObject query = new BasicDBObject();
+            if (title != null) {
+                query.append("title", title);
+            }
+            if (tag != null) {
+                query.append("tags", tag);
+            }
+            MongoCursor<Document> cursor = collection.find(query).skip(pageCount * (pageNumber - 1)).limit(pageCount).iterator();
+            while (cursor.hasNext()) {
+                Document questionDoc = cursor.next();
+                Question question = mapQuestion(questionDoc);
+                list.add(question);
+            }
+            if (list.size() != 0) {
+                return list;
+            }
+            else {
+                return null;
+            }
+        } catch (MongoException e) {
+            logger.warn("Failed to find questions on the database", e);
+            throw new DataAccessException();
+        }
+    }
+
+    @Override
+    public Question saveQuestion(Question question) throws DataAccessException {
+        try{
+            MongoCollection<Document> collection = database.getCollection("questions");
+
+            if (question.getId() == null){
+                question.setId(new ObjectId().toString());
+                question.setCreatedAt(new Date());
+                Document doc = new Document("_id", new ObjectId(question.getId())).append("title", question.getTitle()).append("rate", question.getRate())
+                        .append("content", question.getContent()).append("answers", new ArrayList<Answer>()).append("user", question.getUser())
+                        .append("tags", question.getTags()).append("modified_date", question.getModifiedAt()).append("created_date", question.getCreatedAt());
+                collection.insertOne(doc);
+            }
+            else{
+                BasicDBObject newDocument = new BasicDBObject();
+                newDocument.append("$set", new BasicDBObject().append("title", question.getTitle()).append("rate", question.getRate())
```

Comment:
It's not needed to update the answers field in this query. You just need to update the internal details of the question object. Also, normally the user id who has created the question would not be changed, but there could be a list of editors for the question. You don't need to implement that feature right now, but I wouldn't update the user id here.

Further, rate is not an integer that you take from the input. It is an aggregate field based on user's opinions. So, basically it is a nested dictionary which keeps track of who has rated the question and what was the rate.